include/solvers/gls_sharp_navier_stokes.h

```diff
@@ -57,94 +57,164 @@ class GLSSharpNavierStokesSolver : public GLSNavierStokesSolver<dim>
-
-  void
-  assemble_rhs(const Parameters::SimulationControl::TimeSteppingMethod
-                 time_stepping_method) override;
-
   /**
-   * Members
+  * @brief Calculate the coefficients required for BDF integration of order n.
+  * The coefficients are determined through a recursion algorithm.
+  * The algorithm is taken from : Hay, Alexander, et al. "hp-Adaptive time
+   * integration based on the BDF for viscous flows." Journal of Computational
+   * Physics 291 (2015): 151-176.
+   *
+   * @param order The order of the BDF method. The BDF method of order n requires n+1 arrays
+   *
+   * @param time_steps a vector containing all the time steps. The time steps should be in reverse order.
+   * For example, if the method is a BDF2, it uses three values for the time : (t)
+   * (t-dt_1) and (t-dt_2). Thus the time step vector should contain dt_1 and
+   * dt_2.
    */
+
+    /**
+    * @brief
+    * Map the vertex index to the cell that includes that vertex.
+    * This map is used to find all the cell close to a specific vertex.
+    */
+    void
+    vertices_cell_mapping();
+
+    /**
+    * @brief
+    * Defines the particle structure and value based on the parameter file.
+    */
+    void
+    define_particles();
+
+    /**
+    * @brief
+    * Evaluate the forces applied to each of the IB particles.
+    */
+    void
+    force_on_ib();
+
+    /**
+    * @brief
+    * Modify the system matrix to impose IB condition using the sharp_edge approach.
+    * The detail of this approach are presented in :
+     * L. Barbeau, S. Étienne, C. Béguin & B. Blais,
+     * «Development of a high-order continuous Galerkin sharp-interface immersed boundary method and
+     * its application to incompressible flow problems,» Computers & Fluids, 2020,
+     * in press, ref. CAF-D-20-00773
+    */
+    void
+    sharp_edge();
+
+    /**
+    * @brief
+    * Write in  the output file the forces , velocity , position of each of the
+    * particles.
+    */
+    void
+    write_force_ib();
+
+    /**
+    * @brief
+    * Write in  the output file the forces , velocity , position of each of the
+    * particles.
+    */
+    void
+    integrate_particles();
+
+    /**
+    * @brief
+    * Store the solution of the particles dynamics parameters for integration.
+    * Defines the table to store the history of each of the particles.
+    */
+    void
+    finish_time_step_particles();
+
+    /**
+    * @brief
+    * Evaluate the L2 error on the computational domain if an analytical solutio
+    * is given.
+    */
+    double
+    calculate_L2_error_particles();
+
+
+    virtual void
+    postprocess_fd(bool firstIter) override;
+
+    /**
+    * @brief
+    * Allow a refinement around each of the particles.
+    */
+    void
+    refine_ib();
+
+    /**
+    * @brief
+    * Return a bool to define if a cell is cut by a IB particle, the id of the particle that cut it if it's the case and the local dof of the cell for later use
```

Comment:
```suggestion
    * Return a bool to define if a cell is cut by a IB particle, the id of the particle that cut it if it's the case and the local dof of the cell for later use
```

Apply indentation to the code using the indent-all script. This shouldnt happen.